packages/firebase_analytics/test/observer_test.dart

```diff
@@ -53,6 +69,66 @@ void main() {
 
       verify(analytics.setCurrentScreen(screenName: 'foo')).called(1);
     });
+
+    test('handles only ${PlatformException}s', () async {
+      observer = FirebaseAnalyticsObserver(
+        analytics: analytics,
+        nameExtractor: (RouteSettings settings) => 'foo',
+      );
+
+      final PageRoute<dynamic> route = MockPageRoute();
+      final PageRoute<dynamic> previousRoute = MockPageRoute();
+
+      // Throws non-PlatformExceptions
+      when(analytics.setCurrentScreen(screenName: anyNamed('screenName')))
+          .thenThrow(ArgumentError());
+
+      expect(() => observer.didPush(route, previousRoute), throwsArgumentError);
+
+      // Print PlatformExceptions
+      Future<void> throwPlatformException() async =>
+          throw PlatformException(code: '');
+
+      when(analytics.setCurrentScreen(screenName: anyNamed('screenName')))
+          .thenAnswer((Invocation invocation) => throwPlatformException());
+
+      overridePrint(() => observer.didPush(route, previousRoute));
+
+      await Future<void>.delayed(Duration(seconds: 1));
+      expect(
+        printLog,
+        <String>['$FirebaseAnalyticsObserver: ${PlatformException(code: '')}'],
+      );
+    });
+
+    test('runs onError', () async {
+      bool didRun = false;
+
+      final void Function(PlatformException error) handleError =
+          (PlatformException error) {
+        didRun = true;
+      };
+
+      observer = FirebaseAnalyticsObserver(
+        analytics: analytics,
+        nameExtractor: (RouteSettings settings) => 'foo',
+        onError: handleError,
+      );
+
+      final PageRoute<dynamic> route = MockPageRoute();
+      final PageRoute<dynamic> previousRoute = MockPageRoute();
+
+      Future<void> throwPlatformException() async =>
+          throw PlatformException(code: '');
+
+      when(analytics.setCurrentScreen(screenName: anyNamed('screenName')))
+          .thenAnswer((Invocation invocation) => throwPlatformException());
+
+      observer.didPush(route, previousRoute);
+
+      await Future<void>.delayed(Duration(seconds: 1));
```

Comment:
Can we await on didPush instead of this delayed Future? (same above)

Reply:
Offline discussion: no we can't 😄 (didPush doesn't return a Future), lets just `await Future<void>(() => null);` instead of the delayed future.